packages/babel-generator/src/generators/flow.ts

```diff
@@ -521,11 +522,21 @@ export function TypeAlias(
   this.semicolon();
 }
 
-export function TypeAnnotation(this: Printer, node: t.TypeAnnotation) {
+export function TypeAnnotation(
+  this: Printer,
+  node: t.TypeAnnotation,
+  parent: t.Node,
+) {
   this.token(":");
   this.space();
-  // @ts-expect-error todo(flow->ts) can this be removed? `.optional` looks to be not existing property
-  if (node.optional) this.token("?");
+  if (parent.type === "ArrowFunctionExpression") {
+    this.tokenContext |= TokenContext.arrowFlowReturnType;
+  } else if (
+    // @ts-expect-error todo(flow->ts) can this be removed? `.optional` looks to be not existing property
+    node.optional
+  ) {
+    this.token("?");
```

Comment:
This looks like the case for:

```js

const ast = Babel.packages.parser.parse("type x = ?    number; /**/", {plugins: ['flow']});
Babel.packages.generator.default(ast);
// Outputs: {code: 'type x = ?number; /**/', decodedMap: undefined}
```

(but I can't test it easily through debugging or stepping into the code, since the repl build is helplessly minified - would be nice if a PR preview was readable/debuggable)

Reply:
https://astexplorer.net/#/gist/65e561e94ad90a154646cfd21be92f52/26ebbf2128f469b1469f574a3fb76b167b53f8e2
You can use it.
Also I remember that the repl has source maps, so debugging should be possible. :)

Reply:
@liuxingbaoyu Thank you! Good point, it works very nicely. Somehow I went through `babeljs.io` in source tab and not `output.circle-artifacts.com`... now that I could set a breakpoint I realized my code snippet wasn't for that case (so the comment may be correct lol).

Only strange thing is it's called `optional` in different places too, even though it's just `nullable` (optional in JS means some type or `undefined`).

Reply:
Thanks for the investigation, I'll re-check and see what we need to do here :)